src/Symfony/Component/Yaml/Tests/ParserTest.php

```diff
@@ -66,6 +66,37 @@ public function getDataFormSpecifications()
         return $tests;
     }
 
+    public function testLineBreaksBeforeDocumentSeparatorAndComment()
+    {
+        $yaml = <<<'EOF'
+
+ 
+# Two empty lines above (second - with one whitespace) and two below (first - with three whitespaces).
+   
+
```

Comment:
I think you should rather build this test case using the `\n` escape sequence to make sure that trailing whitespaces don't get lost in the future.